databricks_cli/stack/api.py

```diff
@@ -239,8 +254,61 @@ def _update_job(self, job_settings, job_id):
         :param job_settings: job settings to update the job with.
         :param job_id: physical job_id of job in databricks server.
         """
+
         self.jobs_client.reset_job({'job_id': job_id, 'new_settings': job_settings})
 
+    def _deploy_workspace(self, resource_properties, physical_id, overwrite):
+        """
+        Deploy workspace asset.
+        TODO (alinxie) Change name to overwrite to be more toward overwriting exclusive resources.
+        For a notebook workspace resource, it is recommended to provide 'language' and 'format'
+        :param resource_properties: dict of properties for the workspace asset. Must contain the
+        'source_path' and 'path' fields. The other fields will be inferred if not provided.
+        :param physical_id: dict containing physical identifier of workspace asset on databricks.
+        Should contain the field 'path'.
+        :param overwrite: Whether or not to overwrite the contents of workspace notebooks.
+        :return: (dict, dict) of (physical_id, deploy_output). physical_id is the physical ID for
+        the stack status that contains the workspace path of the notebook or directory on datbricks.
+        deploy_output is the initial information about the asset on databricks at deploy time
+        returned by the REST API.
+        """
+        # Required fields. TODO(alinxie) put in _validate_config
+        local_path = resource_properties.get('source_path')
+        workspace_path = resource_properties.get('path')
+        object_type = resource_properties.get('object_type')
+
+        actual_object_type = 'DIRECTORY' if os.path.isdir(local_path) else 'NOTEBOOK'
+        if object_type != actual_object_type:
+            raise StackError("Field 'object_type' ({}) not consistent"
+                             "with actual object type ({})".format(object_type, actual_object_type))
+
+        click.echo('sync {} {} to {}'.format(click.style(object_type, fg='cyan'),
```

Comment:
Sorry I didn't literally mean color. I meant that we should make this a proper sentence like: `Deploying the <object_type> <local_path> to Databricks at <workspace_path>.`

Reply:
omg... sorry for misunderstanding, will change!